unit_tests/dbp_testing.py

```diff
@@ -140,8 +140,15 @@ def addSkeletonMission(self):
         Assumes self.td has the test/temp directory path (str),
         will populate self.instrument_ids for a list of instruments.
         """
-        dbu = dbprocessing.DButils.DButils(os.path.join(
-            self.td, 'emptyDB.sqlite'))
+        if os.path.exists(os.path.join(self.td, 'emptyDB.sqlite')):
+            dbu = dbprocessing.DButils.DButils(os.path.join(
+                self.td, 'emptyDB.sqlite'))
+        else:
+            self.td = os.getcwd() + "/" + self.td
+            dbu = dbprocessing.DButils.DButils(self.td)
+            for table in reversed(dbu.metadata.sorted_tables):
+                dbu.session.execute(table.delete())
+                dbu.session.commit()
```

Comment:
So I'm reading this as using sqlite if emptyDB exists, but using postgres otherwise? We should probably come up with a better way to switch between the unit tests (environment variable? argument to the test script?). Also the new construction of self.td isn't portable, if the path matters, but it looks like maybe it doesn't. If we're not using sqlite, then probably need a different way to make sure we're working on an empty database (probably have it actually create the db in postgres, if we can manage that....will take some thought.)

Reply:
You cannot create a database with "Create Database".   Since the tables are there, I just empty them.  Actually you don't recreate your "emptyDB.sqlite" file from scratch either, you just copy it.   

I am prety happy about emptying the table between each test, but the switching could be written better.

I don't really remember why I needed to add a "path".  I remember struggling with this.  I think is your "os.path.exist()" which pass the test, but adding the path, I made sure it returned `false`.

Reply:
Looking at context, it's working with the emptyDB as copied into the test directory. self.td is test *directory* not *database*. This code also *deletes* all the tables, rather than emptying them. And then it tries to add columns to tables that no longer exist.

The main difference between what happens with postgres and sqlite here is the construction of the dbu...everything else should be the same. The table deletion/recreation stuff to have an empty database should happen in the setup/teardown.

"Empty" db, incidentally, has the tables in it (it's not a completely empty database with no tables).

Reply:
Actually it does not delete the table but empty them.  I tested that code and looked int the tables.  The database is there with tables after calling this code.  Here is the reference:

https://stackoverflow.com/questions/4763472/sqlalchemy-clear-database-content-but-dont-drop-the-schema

At the beginning I put it in the "addSkeletonMission" since this where you instantiate "dbu".   setup/teardown makes more sense.

Reply:
> Actually it does not delete the table but empty them.

As I realized a couple minutes later :)